### ISSUE TITLE
Sonarr attempts to move torrent files when manually imported from the activity queue under certain conditions

### ISSUE THREAD

**Issue by edrock200** [Sonarr/Sonarr#6737]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

This one is someone hard to explain but...If you use Sonarr's "Wanted" section to manually import something, and chose Move vs Hardlink, then later if you have a "yellow" item in queue that is torrent sourced, and click the manual import icon, map the episodes, and import, Sonarr will attempt to move the torrent files vs hardlink them. Note manually a "yellow" item from the activity queue does not give the "Move/Hardlink-Copy" option in the import dialogue box.

I think whats happening is Sonarr is using the last Move/Hardlink-Copy setting from the "Wanted" section when importing from the queue. I mention this because it's inconsistent, sometimes it works, and I only recently noticed it happened after my last "Wanted" import was set to "Move."

### Expected Behavior

Sonarr always hardlinks/copies torrent files that are manually imported from the activity queue.

### Steps To Reproduce

1. Have your source torrent directory set to read-only
2. Go to wanted, then manual import
3. Select something for manual import not from torrent dir, but a source that has read/write rights
4. Set import type to "Move"
5. Import said item
6. Have sonarr grab a torrent that will have a parsing issue, e.g. will go yellow in the activity queue that is sourced in the read-only torrent source dir
7. Click the manual import item in the activity queue next to the "yellow" torrent
8. Map files and click import
9. Sonarr will act like it's importing, but each import will fail because it cannot move the file. If upgrading existing episodes, it will delete the existing episodes as well, but never import the new ones.
10. To see it work as expected, repeat these steps, but in step 4, set it to "Hardlink/Copy"

### Environment

```markdown
- OS: Ubuntu 18.04
- Sonarr: 4.0.4.1491
- Docker Install: 4.0.4.1491-ls235 by linuxserver.io
- Using Reverse Proxy: Yes, nginx with SSL
- Browser: Chrome Version 123.0.6312.123 (Official Build) (64-bit)
- Database: Sqlite 3.44.2 Migration 205
```

### What branch are you running?

Develop

### Trace Logs?

I was so proud of myself for finally having a proper data set for a bug report, including trace logs. Unfortunately the trace logs appear to have been overwritten since the event and the time of creating this bug. I will attempt to find something to trigger it again and edit with trace, please don't yell at me :)  I was able to pull this out of the events though:

`Couldn't import episode /mnt/imports/torrents/qbit424/make.room.for.daddy/The.Danny.Thomas.Show.S09.1080p.AMZN.WEB-DL.DD2.0.H.264/The.Danny.Thomas.Show.S09E22.Temper,.Temper.1080p.AMZN.WEB-DL.DD2.0.H.264.mkv: Read-only file system : '/mnt/imports/torrents/qbit424/make.room.for.daddy/The.Danny.Thomas.Show.S09.1080p.AMZN.WEB-DL.DD2.0.H.264/The.Danny.Thomas.Show.S09E22.Temper,.Temper.1080p.AMZN.WEB-DL.DD2.0.H.264.mkv'

System.IO.IOException: Read-only file system : '/mnt/imports/torrents/qbit424/make.room.for.daddy/The.Danny.Thomas.Show.S09.1080p.AMZN.WEB-DL.DD2.0.H.264/The.Danny.Thomas.Show.S09E22.Temper,.Temper.1080p.AMZN.WEB-DL.DD2.0.H.264.mkv'
   at System.IO.FileSystem.DeleteFile(String fullPath)
   at System.IO.FileSystem.MoveFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.Move(String sourceFileName, String destFileName, Boolean overwrite)
   at NzbDrone.Mono.Disk.DiskProvider.TransferFilePatched(String source, String destination, Boolean overwrite, Boolean move) in ./Sonarr.Mono/Disk/DiskProvider.cs:line 352
   at NzbDrone.Common.Disk.DiskProviderBase.MoveFile(String source, String destination, Boolean overwrite) in ./Sonarr.Common/Disk/DiskProviderBase.cs:line 250
   at NzbDrone.Common.Disk.DiskTransferService.TryMoveFileVerified(String sourcePath, String targetPath, Int64 originalSize) in ./Sonarr.Common/Disk/DiskTransferService.cs:line 510
   at NzbDrone.Common.Disk.DiskTransferService.TransferFile(String sourcePath, String targetPath, TransferMode mode, Boolean overwrite) in ./Sonarr.Common/Disk/DiskTransferService.cs:line 296
   at NzbDrone.Core.MediaFiles.EpisodeFileMovingService.TransferFile(EpisodeFile episodeFile, Series series, List`1 episodes, String destinationFilePath, TransferMode mode, LocalEpisode localEpisode) in ./Sonarr.Core/MediaFiles/EpisodeFileMovingService.cs:line 170
   at NzbDrone.Core.MediaFiles.EpisodeFileMovingService.MoveEpisodeFile(EpisodeFile episodeFile, LocalEpisode localEpisode) in ./Sonarr.Core/MediaFiles/EpisodeFileMovingService.cs:line 81
   at NzbDrone.Core.MediaFiles.UpgradeMediaFileService.UpgradeEpisodeFile(EpisodeFile episodeFile, LocalEpisode localEpisode, Boolean copyOnly) in ./Sonarr.Core/MediaFiles/UpgradeMediaFileService.cs:line 80
   at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedEpisodes.Import(List`1 decisions, Boolean newDownload, DownloadClientItem downloadClientItem, ImportMode importMode) in ./Sonarr.Core/MediaFiles/EpisodeImport/ImportApprovedEpisodes.cs:line 158
`

### Anything else?

I intentionally have my torrent source directory set to "Read-Only" to prevent accidentally moving torrent files out of the source dir. My torrents and media do not reside on the same volume, however Sonarr has read access to the torrent source volume and full read-write to the media volume where it copies media to.

**Comment by markus101** [comment]
If the files haven't been moved why can't you replicate and post the correct log files?
Sounds more like Sonarr sees the file as complete and finished seeding so it's trying to move the files instead of just copying them, or you managed to queue the import from Sonarr at the exact same time Sonarr removed it from it's queue and has no idea what it is, possible, but the timing would need to be very precise and the manual import from Sonarr would have been pointless because Sonarr would have already been importing them.

For items in the queue the import mode is set at `auto`, which has Sonarr decide based on the information from the download. For testing I use Copy/Hardlink when manually import from Wanted: Missing, but just imported an item from Usenet which would be a move and it copied.

**Comment by edrock200** [comment]
> If the files haven't been moved why can't you replicate and post the correct log files? Sounds more like Sonarr sees the file as complete and finished seeding so it's trying to move the files instead of just copying them, or you managed to queue the import from Sonarr at the exact same time Sonarr removed it from it's queue and has no idea what it is, possible, but the timing would need to be very precise and the manual import from Sonarr would have been pointless because Sonarr would have already been importing them.
> 
> For items in the queue the import mode is set at `auto`, which has Sonarr decide based on the information from the download. For testing I use Copy/Hardlink when manually import from Wanted: Missing, but just imported an item from Usenet which would be a move and it copied.

As of now, I don't have an item in torrents which doesn't auto-parse, so trying to find something that will go "yellow." I do not have Sonarr seed limits set, e.g. it seeds indefinitely. If it does auto parse, it imports correctly (e.g. imports automatically.) It's only when they go yellow and I manually import from activity queue, and my last "wanted" manual import was set to move, that this activity queue import attempts to move vs hardlink/copy if that makes sense.

The manual import piece - I wasn't suggesting that I'm importing the same thing via manual and from the queue. It appears that the last setting used in manual import (Move or Copy), is what is being used when manually importing an item from the queue if that makes sense, even when they are different items.

**Comment by markus101** [comment]
> I do not have Sonarr seed limits set, e.g. it seeds indefinitely.

That doesn't necessarily change whether Sonarr sees something as finished seeding, because usually it's not tracked by those values, but by what your download client reports, though that depends on the client.

> If it does auto parse, it imports correctly (e.g. imports automatically.) It's only when they go yellow and I manually import from activity queue, and my last "wanted" manual import was set to move, that this activity queue import attempts to move vs hardlink/copy if that makes sense.

There was no confusion on my end regarding what you were saying. I was explaining one way you could experience this issue, but it would require you attempting to do two things at virtually the same time.

> The manual import piece - I wasn't suggesting that I'm importing the same thing via manual and from the queue.

I'm not saying that you did, I'm saying I do and as a counter point to you saying it was using Move, I was saying mine did not sure Copy/Hardlink even though that's the last value selected.

Once there are trace logs that cover the issue we should know more.

**Comment by edrock200** [comment]
Was finally able to reproduce @markus101 Please look around the 13:38 time markers for 90 Day - Pillow Talk - S25E15 - Tell All Part 4 WEBDL-1080p.mkv
[sonarr.trace.31.txt](https://github.com/Sonarr/Sonarr/files/15048048/sonarr.trace.31.txt)

**Comment by markus101** [comment]
We also need to see the information on the torrents from Qbit, also in the trace logs.

> 2024-04-20 13:38:04.7|Trace|ManualImportService|Processing file 16 of 16

Did the other episodes fail to import with the exact same reason?

**Comment by edrock200** [comment]
> We also need to see the information on the torrents from Qbit, also in the trace logs.
> 
> > 2024-04-20 13:38:04.7|Trace|ManualImportService|Processing file 16 of 16
> 
> Did the other episodes fail to import with the exact same reason?

Yes they did. I'll try to reproduce and grab trace logs from both sides. Thanks again.

**Comment by edrock200** [comment]
@markus101 I apologize for wasting your time. I think I know what's happening. After trying to reproduce, I noticed something odd. Things disappearing from queue. I have a family member who also has access and when I inquired with him I found out something he's been doing that is most likely causing this. When an item is in queue as yellow from a torrent site, he was manually importing/mapping and kicking off the import, then clicking the x and selecting change category. It appears as soon as the category changes, sonarr switches from hardlink/copy to move. He told me he was doing this so they dont stay in queue while importing, so if I happened to see it I wouldn't attempt to import it again as well. 

Had I known this, I wouldn't have wasted your time. Apologies again.
